### PR TITLE
fixed #2755?,#2756,#2762,#2596 and #2738

### LINKED ISSUE #2755: In IE9 all content under the table is increasingly pushed down on hover of an x/mref
select typeTest

hover over an x/mref link
- the scrollbar will change, and content under the table will be pushed down, an increasing region of whitspace will be shown...

### LINKED ISSUE #2756: Using multiple annotators will result in an error
Multiple annotators results in: "Removing of existing attributes is currently not supported."

Probably because the metadata from the original entity is used instead of the resulted metadata from the previous annotator

### PR DESCRIPTION
**Pull request by bartcharbon**
Discuss for #2738 how to deal with "COUNT" permissions on a plugin, should this even be possible?

### CODE REVIEW

File: molgenis-data-elasticsearch/src/main/java/org/molgenis/data/elasticsearch/ElasticsearchRepositoryCollection.java [hunk 1]
```diff
@@ -92,11 +92,12 @@ public void deleteEntityMeta(String entityName)
 	@Override
 	public void addAttribute(String entityName, AttributeMetaData attribute)
 	{
-		EntityMetaData entityMetaData = dataService.getEntityMetaData(entityName);
+		EntityMetaData entityMetaData = dataService.getMeta().getEntityMetaData(entityName);
```
Comment (fdlk) [inline molgenis-data-elasticsearch/src/main/java/org/molgenis/data/elasticsearch/ElasticsearchRepositoryCollection.java]: The above change, I understand, but this one I find a little worrisome. Does it mean that after annotating the dataService thinks the repo has different metadata than the metadataservice thinks it has?

Comment (fdlk) [inline molgenis-data-elasticsearch/src/main/java/org/molgenis/data/elasticsearch/ElasticsearchRepositoryCollection.java]: shouldn't the repository get updated with the new metadata? or does that happen later?

Comment (bartcharbon) [inline molgenis-data-elasticsearch/src/main/java/org/molgenis/data/elasticsearch/ElasticsearchRepositoryCollection.java]: See 4 lines below, but I'll rewrite that bit anyhow, was planning to but forgot.
The change here is probably unnecessary, a artefact of a attempt to understand/fix what was going on.

File: molgenis-data/src/main/java/org/molgenis/data/meta/MetaDataServiceImpl.java [hunk 2]
```diff
@@ -287,6 +287,7 @@ public void refreshCaches()
 	@Transactional
 	public List<AttributeMetaData> updateSync(EntityMetaData sourceEntityMetaData)
 	{
+
```
Comment (Mark-de-Haan) [inline molgenis-data/src/main/java/org/molgenis/data/meta/MetaDataServiceImpl.java]: Didnt know java had white space coding ;p

Comment (fdlk) [inline molgenis-data/src/main/java/org/molgenis/data/meta/MetaDataServiceImpl.java]: The devil is in the details

File: molgenis-data/src/main/java/org/molgenis/data/support/MapEntity.java [hunk 3]
```diff
@@ -61,7 +61,8 @@ public void set(Entity other, EntityMetaData metaData)
 		this.idAttributeName = entityMetaData.getIdAttribute().getName();
 		for (AttributeMetaData attribute : metaData.getAtomicAttributes())
 		{
-			set(attribute.getName(), other.get(attribute.getName()));
+			if (Iterables.contains(other.getAttributeNames(), attribute.getName())) set(attribute.getName(),
```
Comment (fdlk) [inline molgenis-data/src/main/java/org/molgenis/data/support/MapEntity.java]: This change is what breaks the annotator unit tests. `other` is a mock and getAttributeNames returns null.

**Pull request merged**